js/ui/viewSelector.js

```diff
@@ -346,6 +346,14 @@ const ViewSelector = new Lang.Class({
 
         let symbol = event.get_key_symbol();
 
+        if (this._activePage == this._workspacesPage) {
+            if (symbol == Clutter.Escape) {
+                Main.overview.toggleWindows();
+                return true;
+            }
+            return false;
```

Comment:
It's probably fine, but I liked to think of this block as specific to the workspace page, where the `!global.stage.key_focus` statement that comes afterwards does not make sense.